RecoCTPPS/PixelLocal/src/RPixClusterToHit.cc

```diff
@@ -0,0 +1,117 @@
+#include "RecoCTPPS/PixelLocal/interface/RPixClusterToHit.h"
+
+
+RPixClusterToHit::RPixClusterToHit(edm::ParameterSet const& conf)
+{
+verbosity_ = conf.getUntrackedParameter<int>("RPixVerbosity");
+}
+
+RPixClusterToHit::~RPixClusterToHit(){}
+
+void RPixClusterToHit::buildHits(unsigned int detId, const std::vector<CTPPSPixelCluster> &clusters, std::vector<CTPPSPixelRecHit> &hits)
+{
+  
+  if(verbosity_) edm::LogInfo("RPixClusterToHit")<<" RPixClusterToHit "<<detId<<" received cluster array of size = "<<clusters.size();
+   for(unsigned int i=0; i<clusters.size();i++){
+     make_hit(clusters[i],hits);
+  }
+
+}
+
+
+void RPixClusterToHit::make_hit(CTPPSPixelCluster aCluster,  std::vector<CTPPSPixelRecHit> &hits ){
+
+// take a cluster, generate a rec hit and push it in the rec hit vector
+
+//call the topology
+  CTPPSPixelSimTopology topology;
+//call the numbering inside the ROC
+  CTPPSPixelIndices pxlInd;
+// get information from the cluster 
+// get the whole cluster size and row/col size
+  unsigned int thisClusterSize = aCluster.size();
+  unsigned int thisClusterRowSize = aCluster.sizeRow();
+  unsigned int thisClusterColSize = aCluster.sizeCol();
+
+// get the minimum pixel row/col
+  unsigned int thisClusterMinRow = aCluster.minPixelRow();
+  unsigned int thisClusterMinCol = aCluster.minPixelCol();
+
+// calculate "on edge" flag
+  bool anEdgePixel = false;
+  if(aCluster.minPixelRow() == 0 || aCluster.minPixelCol() == 0 ||  
+     int(aCluster.minPixelRow()+aCluster.rowSpan()) == (pxlInd.getDefaultRowDetSize()-1) ||  
+     int(aCluster.minPixelCol()+aCluster.colSpan()) == (pxlInd.getDefaultColDetSize()-1) ) 
+    anEdgePixel = true;
+
+// check for bad (ADC=0) pixels in cluster 
+  bool aBadPixel = false;
+  for(unsigned int i = 0; i < thisClusterSize; i++){
+    if(aCluster.pixelADC(i)==0) aBadPixel = true;
+  } 
+
+// check for spanning two ROCs
+  bool twoRocs = false;
+  int currROCId = pxlInd.getROCId(aCluster.pixelCol(0),aCluster.pixelRow(0) ); 
+  if(thisClusterSize>1)
```

Comment:
This check is useless here, as it is implicit in the following loop instructions

Reply:
ok. removed.